sympy/parsing/mathematica.py

```diff
@@ -1,133 +1,433 @@
 from __future__ import print_function, division
 
+from collections import defaultdict
+from itertools import product
 import re
 from sympy import sympify
 
 
-def mathematica(s):
-    return sympify(parse(s))
+class mathematica(object):
+    '''This class converts a string of a basic Mathematica expretion to
+    SymPy style. Output is a SymPy object.'''
+
+    # use this class like a function
+    def __new__(cls, s, additional_translation={}):
+        try:
+            # get parsed string
+            s = cls.parse(s, additional_translation)
```

Comment:
Can this line ever be run? `__new__` never returns an instance of this class.

Reply:
In this code, I wanted `mathematica` class to behave like a function, instead of creating an instance.